lib/services/secure-submit/hps-credit-service.js

```diff
@@ -50,23 +50,27 @@ function HpsCreditService(hpsConfig, soapUri) {
      * @return {Object} exports for chaining
      */
     var chargeWithCard =
-        function chargeWithCard(amount, currency, card, cardHolder, requestMultiUseToken, memo, callback) {
+        function chargeWithCard(amount, currency, card, cardHolder, requestMultiUseToken, additionalTxnFields, callback) {
             var schema = porticoSchema.requestType('CreditSale'),
                 tx = {};
 
             if (hlp.defNn(card)) {
                 tx.CardData = {};
-                tx.CardData.ManualEntry = hlp.hydrateCardData(card);
+                if (typeof(card)==='string'){
+                    tx.CardData.TrackData = card;
```

Comment:
For track data, we'll need to have the ability to set the `method` attribute of the `TrackData` data field as either `swipe` or `proximity` to allow for contactless as well, so it may be better to pass the track data as an object similar to our other SDKs. You can reference our .NET SDK: [Data Class](https://github.com/hps/heartland-dotnet/blob/master/SecureSubmit/Entities/HpsTrackData.cs), [Hydration](https://github.com/hps/heartland-dotnet/blob/master/SecureSubmit/Services/HpsSoapGatewayService.cs#L139-L145).

Also, our gateway supports encrypted track data (a large portion of the readers are encrypted with Heartland's End-To-End Encryption [E3]), so it would be nice to have this capability as well. You can reference our .NET SDK: [Data Class](https://github.com/hps/heartland-dotnet/blob/master/SecureSubmit/Entities/HpsEncryptionData.cs), [Hydration](https://github.com/hps/heartland-dotnet/blob/master/SecureSubmit/Services/HpsSoapGatewayService.cs#L121-L128).